### PR TITLE
Crosswords check other clues on clear

### PR DESCRIPTION
**Pull request by NathanielBennett**
"Is it worth it?" (Elvis Costello)

Deals with clearing intersecting letters: Checks each entry in a clue to see whether it intersects with other completed entries and only clears the correct letters. 

The jesus clue(Crypic 26666)
before-clear

onClear before: 
clear-before

onClear after: 
clear-after

### CODE REVIEW

**Comment by crifmulholland** [comment]
Beautiful!

**Comment by ScottPainterGNM** [comment]
This guy thinks you are a legend. Great spot!

image

**Comment by desbo** [comment]
impressive tests, nice one! :+1:

File: static/src/javascripts/test/spec/common/crosswords/helpers.spec.js [hunk 1]
```diff
@@ -243,4 +244,253 @@ describe('Helpers', function () {
             expect(helpers.cellsForClue(entriesFixture, clue)).toEqual(expectedCells);
         });
     });
+
+    describe('areCluesInAGroup', function () {
+        var thisEntryFixture = {
```
Comment (jbreckmckye) [inline static/src/javascripts/test/spec/common/crosswords/helpers.spec.js]: Given that we can run ES6 tests, should we start using `const` and `let` in them wherever possible?

**Pull request merged**